src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let resultSum = 0;
+
+  const funAdder = (a) => {
+    if (a === undefined) {
+      const returnValue = resultSum;
+
+      resultSum = 0;
+
+      return returnValue;
+    } else {
+      resultSum += a;
+
+      return funAdder;
```

Comment:
The `funAdder` function correctly implements the logic for adding numbers or returning the sum when called without arguments. However, you should consider using a more descriptive parameter name instead of `a`. While it is clear from the context that `a` represents the number to add, using a more descriptive name such as `numberToAdd` would improve code readability.